base/show.jl

```diff
@@ -748,22 +748,17 @@ function show_unionaliases(io::IO, x::Union)
 end
 
 function show(io::IO, ::MIME"text/plain", @nospecialize(x::Type))
-    show(io, x)
-    if !print_without_params(x) && get(io, :compact, true)
+    if !print_without_params(x)
         properx = makeproper(io, x)
         if make_typealias(properx) !== nothing || x <: make_typealiases(properx)[2]
```

Comment:
```suggestion
        if make_typealias(properx) !== nothing || (unwrap_unionall(x) isa Union && x <: make_typealiases(properx)[2])

```

Otherwise you also get this other funny bug:
```
julia> @eval Base const VKD5{T} = Dict{T, Integer}; @eval Base export VKD5; VKD5
VKD5{<:Any} (alias for Dict{T, Integer} where T)

julia> @eval Base const VKD4{T, Integer} = Dict{T}; @eval Base export VKD4; VKD4
VKD4{<:Any, <:Any} (alias for Dict{T, V} where V where T)

julia> VKD5
Dict{T, Integer} where T (alias for Dict{T, Integer} where T)
```

Reply:
Hm, seems there's already other problems with this PR (see doctests), so I'll fix that in a separate PR instead.